source/class/qx/ui/form/IListItem.js

```diff
@@ -20,4 +20,51 @@
  * This is a marker interface for classes which can act as a children of
  * {@link qx.ui.form.List}
  */
-qx.Interface.define("qx.ui.form.IListItem", {});
+qx.Interface.define("qx.ui.form.IListItem", {
+    /*
+  *****************************************************************************
+     EVENTS
+  *****************************************************************************
+  */
+
+  events: {
+    /** Fired when the value was modified */
+    changeReadOnly: "qx.event.type.Data"
+  },
+
+  /*
+  *****************************************************************************
+     MEMBERS
+  *****************************************************************************
+  */
+
+  members: {
+    /*
+    ---------------------------------------------------------------------------
+      READONLY PROPERTY
+    ---------------------------------------------------------------------------
+    */
+
+    /**
+     * Sets the element's value.
+     *
+     * @param value {Boolean|null} The new value of the element.
+     * @return {Boolean|null}
+     */
+    setReadOnly(value) {
+      return arguments.length == 1;
```

Comment:
I don't understand this line, is this an Interface-specific check?

Reply:
Me neither, I just copied it from qx.ui.form.IBooleanForm, seems to be used in some interfaces.

Reply:
There is `qx.core.Assert.assert` statement for this goal. Not sure returning from interface method will work.

Reply:
> I don't understand this line, is this an Interface-specific check?

in theory interface members provide some validation of the parameters, given that the generator was unable to do checks (the compiled could, but doesn't).  But it's quite random whether its used or not

Reply:
I recalled that the return value from a function in an Interface could be used for validation, but I can't find that documented anywhere and there are no tests to ensure that method of validation continues to work. There are a couple of tests where the Interface method returns `true`, but none it returns `false`, and the meaning of the return value is entirely undocumented. The only documented way of doing parameter validation, it seems, is with the various `assert*` methods. The use of a boolean return value, as used here (and elsewhere) should really be replaced with a call to one of the `assert` methods.

Reply:
And I've now confirmed in the code that the return value from the "precondition" function is ignored:

Interface.js:525
```
          // call precondition
          preCondition.apply(this, arguments);
```

Reply:
Seems to have been changed 15 years ago ;-)  https://github.com/qooxdoo/qooxdoo/commit/84c789da40aa3e73919de5f461dd3b20bef75f37

The documentation in interface still describes the old behaviour and needs to be changed too. Also all Interfaces with return values should be changed to use assertions instead. But that sould be done on another PR.

Reply:
Ah! They say that memory is the second thing to go -- I can't recall what the first is... but I did recall correctly. :-)